internal/elements.go

```diff
@@ -390,6 +390,19 @@ func (etag ETag) String() string {
 	return fmt.Sprintf("%q", string(etag))
 }
 
+// https://datatracker.ietf.org/doc/html/rfc2068#section-14.25 and
+// https://datatracker.ietf.org/doc/html/rfc2068#section-14.26
+func ParseConditionalHeader(hdr string) (string, error) {
+	if hdr == "" || hdr == "*" {
+		return hdr, nil
```

Comment:
The ETag can be `"*"`, in which case this function can't differentiate between that and a wildcard.

Maybe the wildcard checking is best left to the caller, and this function should be called `ParseETag`.

Reply:
A very good point. Unfortunately, this makes the current API insufficient. As it is external applications (backends) that will have to deal with this, I am proposing to export a type to help them deal with this (see updated change set).